lib/assets/javascripts/twine.coffee

```diff
@@ -124,7 +125,9 @@
     elements[node.bindingId] ?= {}
 
   # Queues a refresh of the DOM, batching up calls for the current synchronous block.
-  Twine.refresh = ->
+  # The callback will be called once when the refresh has completed.
+  Twine.refresh = (callback) ->
+    refreshCallbacks.push(callback) if callback
     return if refreshQueued
     refreshQueued = true
     setTimeout(Twine.refreshImmediately, 0)
```

Comment:
Edit: never mind me, `refreshImmediately` is synchronous anyway

![gif-keyboard-14694901250535498836](https://cloud.githubusercontent.com/assets/1802885/20318420/302780ac-ab38-11e6-881b-151a924896e2.gif)